hibernate-core/src/main/java/org/hibernate/SynchronizeableQuery.java

```diff
@@ -46,8 +46,8 @@
 	 */
 	default SynchronizeableQuery<T> addSynchronizedQuerySpace(String... querySpaces) {
 		if ( querySpaces != null ) {
-			for ( int i = 0; i < querySpaces.length; i++ ) {
-				addSynchronizedQuerySpace( querySpaces[i] );
+			for (String querySpace : querySpaces) {
```

Comment:
We prefer the form you changed from.  This form can have performance impact and for no real gain